zebra/zebra_mpls.c

```diff
@@ -227,8 +249,10 @@ static int lsp_install(struct zebra_vrf *zvrf, mpls_label_t label,
 			changed++;
 		} else {
 			/* Add LSP entry to this nexthop */
-			nhlfe = nhlfe_add(lsp, lsp_type, nexthop->type,
-					  &nexthop->gate, nexthop->ifindex,
+			nhlfe = nhlfe_add(lsp, lsp_type,
+					  nexthop_via_recursive->type,
+					  &nexthop->gate,
+					  nexthop_via_recursive->ifindex,
 					  nexthop->nh_label->num_labels,
 					  nexthop->nh_label->label,
```

Comment:
trying to be clearer about the question here: so this says "send an mpls packet with the LU labels on a link where there's no assurance that those labels mean anything. and the destination mac will be ... for an address that doesn't exist on the link." so ... how can that be meaningful to do?